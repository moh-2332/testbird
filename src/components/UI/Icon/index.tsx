import React, { useState } from 'react';

import classes from './Icon.module.scss'
import Tooltip from '../Tooltip';

interface IconProps {
    children?: any;
    className?: string;
    tooltipText?: string;
    onClick?: () => void;
}

const Icon: React.FC<IconProps> = ({ children, className, tooltipText, onClick }) => {
    const [showTooltip, setShowTooltip] = useState(false);

    return (
        <div
            className={`${classes.icon} ${className ? className : ''}`}
            onClick={() => onClick && onClick()}
            onMouseOver={() => setShowTooltip(true)}
            onMouseOut={() => setShowTooltip(false)}
        >
            {children}
            {showTooltip && tooltipText && <Tooltip text={tooltipText} />}
        </div >
    );
}

export default Icon;