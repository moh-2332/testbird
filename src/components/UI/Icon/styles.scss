.icon {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 40px;
    height: 40px;
    border-radius: 50%;
    position: relative;

    img {
        width: 100%;
        height: 100%;
        border-radius: 50%;
    }
}