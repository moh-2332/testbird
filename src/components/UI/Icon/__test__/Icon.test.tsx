import React from "react";

import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";

import Icon from "../";

const handler = jest.fn();

describe("Icon tests", () => {
    it("Should match the snapshot", () => {
        const homeIcon = renderer.create(<Icon><span className="material-icons">home</span></Icon>);
        expect(homeIcon).toMatchSnapshot();
    });

    it("Should renders correctly", () => {
        render(<Icon><span className="material-icons">home</span></Icon>);

        const homeIcon = screen.getByTestId("icon");
        expect(homeIcon).toBeInTheDocument();
    });

    it("Should run the onclick function", () => {
        render(<Icon onClick={handler}><span className="material-icons">home</span></Icon>);

        const homeIcon = screen.getByTestId("icon");
        userEvent.click(homeIcon);

        expect(handler).toBeCalled();
    });

    it("Should show/hide the tooltip after hovering/unhovering", () => {
        render(<Icon tooltipText="home"><span className="material-icons">home</span></Icon>);

        const homeIcon = screen.getByTestId("icon");
        userEvent.hover(homeIcon);

        const tooltip = screen.getByTestId("tooltip");
        expect(tooltip).toBeInTheDocument();

        userEvent.unhover(homeIcon);

        expect(tooltip).not.toBeInTheDocument();
    });
});