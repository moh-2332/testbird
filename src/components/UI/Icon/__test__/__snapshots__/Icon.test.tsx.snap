// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon tests Should match the snapshot 1`] = `
<div
  className="icon "
  data-testid="icon"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <span
    className="material-icons"
  >
    home
  </span>
</div>
`;
